content/docs/reference-react-dom-server.md

```diff
@@ -40,47 +40,91 @@ The following methods can be used in the environments that don't support streams
+      // The stream can be safely ignored.
+      res.statusCode = 500;
+      res.send(
+        '<!doctype html><p>Loading...</p><script src="clientrender.js"></script>'
+      );
+    }
     onShellReady() {
       // The content above all Suspense boundaries is ready.
       // If something errored before we started streaming, we set the error code appropriately.
       res.statusCode = didError ? 500 : 200;
       res.setHeader('Content-type', 'text/html');
       stream.pipe(res);
     },
-    onShellError(error) {
-      // Something errored before we could complete the shell so we emit an alternative shell.
-      res.statusCode = 500;
-      res.send(
-        '<!doctype html><p>Loading...</p><script src="clientrender.js"></script>'
-      );
+    onError(err) {
+      // TODO: when is this called?
+      // ??? Called if one of the suspended content blocks failed to render.
+      didError = true;
+      console.error(err);
     },
-    onAllReady() {
-      // If you don't want streaming, use this instead of onShellReady.
-      // This will fire after the entire page content is ready.
-      // You can use this for crawlers or static generation.
+  }
+);
+```
 
-      // res.statusCode = didError ? 500 : 200;
-      // res.setHeader('Content-type', 'text/html');
-      // stream.pipe(res);
-    },
+#### Suspense approach two: Wait until all suspended blocks are ready
+
+If you’d like to support clients that can’t run client-side JavaScript like crawlers, you can wait until all suspended blocks are ready and then stream the HTML in one go.
+
+To do so, implement the `onAllReady` callback and start streaming your response here. The `onError` option will be called if an error occurs during rendering.
+
+```javascript
+let didError = false;
+const stream = renderToPipeableStream(
+  <App />,
+  {
     onError(err) {
+      // Called if an error occurred while rendering.
       didError = true;
       console.error(err);
     },
+    onAllReady() {
+      // If you don't want streaming, use this instead of onShellReady.
+      // This will fire after the entire page content is ready.
+      // You can use this for crawlers or static generation.
+      res.statusCode = didError ? 500 : 200;
+      res.setHeader('Content-type', 'text/html');
+      stream.pipe(res);
+    },
   }
 );
 ```
 
-See the [full list of options](https://github.com/facebook/react/blob/14c2be8dac2d5482fda8a0906a31d239df8551fc/packages/react-dom/src/server/ReactDOMFizzServerNode.js#L36-L46).
+#### Hydrating to add interactivity {#rendertopipeablestream-hydrate}
+
+If you call [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on a DOM node that already has this server-rendered markup, React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience. This works with either approach to handling suspense.
+
+#### Options
+
+- `identifierPrefix?: string`
```

Comment:
I would be great to document what at least some of these are for, or to link to somewhere that shows how to implements them correctly. But I think at least putting them here is a good first step?